File: frontend/src/types/doctor.ts
export interface Doctor {
  id: number;
  firstName: string;
  lastName: string;
  specialization: string;
  phoneNumber: string;
  email: string;
  schedule: {
    day: string;
    startTime: string;
    endTime: string;
  }[];
  createdAt: string;
  updatedAt: string;
}

export interface DoctorFormData {
  firstName: string;
  lastName: string;
  specialization: string;
  phoneNumber: string;
  email: string;
  schedule: {
    day: string;
    startTime: string;
    endTime: string;
  }[];
} 

File: backend/models/doctor.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Boolean
from sqlalchemy.orm import relationship
from extensions import db

class Doctor(db.Model):
    __tablename__ = 'doctors'

    id = Column(Integer, primary_key=True)
    first_name = Column(String(50), nullable=False)
    last_name = Column(String(50), nullable=False)
    specialization = Column(String(100), nullable=False)
    phone = Column(String(15), nullable=False)
    email = Column(String(100), nullable=False)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Add relationships
    prescriptions = relationship('Prescription', back_populates='doctor', cascade='all, delete-orphan')
    lab_tests = relationship('LabTest', back_populates='doctor', cascade='all, delete-orphan')

    def to_dict(self):
        return {
            'id': self.id,
            'first_name': self.first_name,
            'last_name': self.last_name,
            'specialization': self.specialization,
            'phone': self.phone,
            'email': self.email,
            'is_active': self.is_active,
            'created_at': self.created_at.isoformat(),
            'updated_at': self.updated_at.isoformat()
        } 